### ISSUE TITLE
Is it possible to use this without langchain?

### ISSUE THREAD

**Issue by ntindle** [NVIDIA/NeMo-Guardrails#30]
I have a specific use case that would be enabled, but it doesn't support langchain and probably won't for the foreseeable future.

It seems this has a hard dependency on langchain in the requirements.txt, but I'd like to know if that can be removed 
or swapped out.

**Comment by drazvan** [comment]
Hi @ntindle! What do you mean by "it doesn't support langchain"? Are you saying that for your use case you are not allowed to have `langchain` as a dependency?

**Comment by ntindle** [comment]
Yeah exactly! We're working on a project that can't use langchain as a dependency

**Comment by ntindle** [comment]
Let me clarify a bit, Auto-GPT isn't built to use langchain's llm functionalities, but does use other functionalities from langchain such as the tools

**Comment by Pouyanpi** [comment]
Hi @ntindle is this still something you'd be interested in? Your input might put weight on how we are going to evolve NeMo Guardrails in near future. We might want to do some refactoring to decouple some of these dependencies. What do you think?

**Comment by ntindle** [comment]
I would appreciate it :) 

We ended up not being able to use it because of the couplings, but we would love to be able to add this as a supported feature set. 

Would you like to have a chat about how we handle different LLM providers or something similar?

**Comment by ntindle** [comment]
Cc @aarushik93 for context on adding guardrails discussion from conference

**Comment by aarushik93** [comment]
@Pouyanpi hello, jumping in here for Nick - do you have a contact so we can set up some time for a discussion?

**Comment by whisper-bye** [comment]
replace langchain with aisuite
https://github.com/andrewyng/aisuite

**Comment by kamilmuratyilmaz** [comment]
I realize it has been 1 year but it is possible to use without langchain now? The langchain framework causing issues a lot in our case and usually breaks stability in long term. A framework like aisuite or support for custom made frameworks would be great to see in nemoguardrails. We are using guardrails more than 1 year and most of our issues related with langchain but not the nemoguardrails itself.

**Comment by Pouyanpi** [comment]
@kamilmuratyilmaz I’m working on this and gradually decoupling from LangChain. The earliest we expect to complete this is by the end of Q1 2026.

**Comment by kamilmuratyilmaz** [comment]
@Pouyanpi Thank you for in advance info. Please let me know if there is anything I can contribute during my spare time.

**Comment by Pouyanpi** [comment]
## LangChain Decoupling (PR Tracker)

 - #1745
  - #1759
  - #1760
  - #1769
  - #1770
  - #1772
  - #1773
  - #1789
  - #1790
  - #1797
  - #1801
  - #1806
  - #1828
  - #1829
  - #1830 
  - #1840

**Comment by Pouyanpi** [comment]
Finally, this is complete! doc updates will follow ahead of the upcoming 0.22.0 release.

cc @ntindle @aarushik93 @kamilmuratyilmaz
